account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts

```diff
@@ -0,0 +1,82 @@
+import { encodeAbiParameters, type Address, type Hex } from "viem";
+
+import { timeRangeModuleAbi } from "./abis/timeRangeModuleAbi.js";
+
+const addresses = {
+  default: "0x00", // TO DO: replace with deployed module address?
+} as Record<number | "default", Address>;
+
+const meta = {
+  name: "TimeRangeModule",
+  version: "alpha.1",
+  addresses,
+};
+
+// Todo: some unified type for ERC-6900 v0.8 modules. I couldn't figure out how to parameterize the class itself over the abi type parameters for onInstall and onUninstall.
```

Comment:
What might work:
```ts
type Module<TAbi extends Abi, InstallData, UninstallData> = {
  meta: ModuleMeta,
  abi: TAbi,
  encodeOnInstallData: (args: InstallData) => Hex;
  encodeOnUninstallData: (args: UninstallData) => Hex;
};
```
Let me know if you want to jam on this.

Reply:
created an [Asana ticket](https://app.asana.com/0/1205598840815267/1209002631957336/f) to address this for the next iteration of the MAv2 SDK!